File: dashboard/src/types/guardrails.ts
import type { RiskLevel } from "@/types/ec2"

export type GuardrailDecision = {
  resource: string
  originalRecommendation: string
  finalDecision: string
  outcome: "Kept" | "Changed" | "Blocked"
  risk: RiskLevel
  blastRadius: string
  reason: string
}

File: dashboard_backend/app/schemas/guardrails.py
from typing import Literal

from pydantic import BaseModel


class GuardrailDecisionResponse(BaseModel):
    resource: str
    originalRecommendation: str
    finalDecision: str
    outcome: Literal["Kept", "Changed", "Blocked"]
    risk: Literal["Low", "Medium", "High", "Critical"]
    blastRadius: str
    reason: str
